bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go

```diff
@@ -508,6 +519,7 @@ func (r *KubeadmConfigReconciler) joinWorker(ctx context.Context, scope *Scope)
 	if res, err := r.reconcileDiscovery(ctx, scope.Cluster, scope.Config, certificates); err != nil {
 		return ctrl.Result{}, err
 	} else if !res.IsZero() {
+		log.Info("Reconcile finished - discovery configuration set")
```

Comment:
I will drop this in favor of logging inside reconcileDiscovery when we return non zero values